### ISSUE TITLE
DH_set0_key wrongly checks for non-NULL arguments

### ISSUE THREAD

**Issue by oerdnj** [libressl-portable/openbsd#92]
The compatibility function `DH_set0_key` added in LibreSSL 2.7.0 has a different (possibly unwanted) behaviour from OpenSSL 1.1.0.  In OpenSSL, it is perfectly OK to add just add pub_key to otherwise empty DH structure, e.g. this is OK:

```
dh = DH_new();
BIGNUM pub_key;
BN_hex2bn(pub_key, "02");
DH_set0_key(dh, pub_key, NULL);
```

In LibreSSL, this would fail to set the internal structures because `dh->priv_key == NULL`.

I don't think having quirks like this is desirable for users of both libraries.

**Comment by 4a6f656c** [comment]
Thanks for the report - this is a bug and will be addressed shortly.

**Comment by botovq** [comment]
Fixed in LibreSSL 2.7.3 that was released a few hours ago. Thanks again for the report.
